### PR TITLE
ci prepare-build-cache

### LINKED ISSUE #21102: Breeze: Add 'prepare-image-cache'  in Breeze
We have a separate command that prepares image caches. It is very similar to "building image" but:
* it should prepare both prod and CI images
* the `docker build` command should be slighly modified (--cache-to and some other commands)
* validation on whether the `buildx plugin` needs to be performed (and command should fail if not with helpful message)

Those "differences" between standard build-image can be found with `PREPARE_BUILDX_CACHE` variable == "true" usage in old breeze

### PR DESCRIPTION
**Pull request by Bowrna**
closes : #21102
related: #21102

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by Bowrna** [comment]
@potiuk 
`--cache:max should be added also in the CI image (because it is multi-segment image now)`
https://github.com/apache/airflow/pull/22492#issuecomment-1076650709

I will add this as part of this PR.

**Comment by Bowrna** [comment]
@potiuk i want to know how we can pass the `CI` flag in python? In bash we set it as env var. Do we need to pass that as flag when we rewrite in python?

https://github.com/apache/airflow/blob/c58b5e583bcad83048fdee72bc9ddf6c3a7b4229/scripts/ci/libraries/_build_images.sh#L454-L463

**Comment by Bowrna** [comment]
@potiuk I made changes specified in this https://github.com/apache/airflow/pull/22618 
Please verify it

**Comment by Bowrna** [comment]
> @potiuk i want to know how we can pass the `CI` flag in python? In bash we set it as env var. Do we need to pass that as flag when we rewrite in python?
> 
> https://github.com/apache/airflow/blob/c58b5e583bcad83048fdee72bc9ddf6c3a7b4229/scripts/ci/libraries/_build_images.sh#L454-L463

@potiuk Please share your view on this when you get time.

File: dev/breeze/src/airflow_breeze/breeze.py [hunk 1]
```diff
@@ -302,7 +303,8 @@ def build_ci_image(
     build_cache: Optional[str],
     platform: Optional[str],
     debian_version: Optional[str],
-    upgrade_to_newer_dependencies: Optional[str],
+    upgrade_to_newer_dependencies: bool,
```
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/breeze.py]:
```suggestion
    upgrade_to_newer_dependencies: bool,
```
This should be Optional[str] actually :)

Comment (potiuk) [reply dev/breeze/src/airflow_breeze/breeze.py]: That was the fix I made here -> https://github.com/apache/airflow/pull/22597

Comment (Bowrna) [reply dev/breeze/src/airflow_breeze/breeze.py]: @potiuk Is it Optional[str] for prod image too?

Comment (potiuk) [reply dev/breeze/src/airflow_breeze/breeze.py]: Yep

**Comment by potiuk** [comment]
> > @potiuk i want to know how we can pass the `CI` flag in python? In bash we set it as env var. Do we need to pass that as flag when we rewrite in python?
> > https://github.com/apache/airflow/blob/c58b5e583bcad83048fdee72bc9ddf6c3a7b4229/scripts/ci/libraries/_build_images.sh#L454-L463
> 
> @potiuk Please share your view on this when you get time.

Same as in Bash. Just add the ``PIP_PROGRESS_BAR`` if CI is set. There is no need to pass it further. But in order to do it nicely you can add ``--ci`` parameter with CI `envvar` via click.

File: dev/breeze/src/airflow_breeze/breeze.py (hunk 1 above)
Comment (potiuk) [reply dev/breeze/src/airflow_breeze/breeze.py]: I added more fixes on the parameter in #22649 :). You will need to rebase and resolve conflicts :)

**Comment by Bowrna** [comment]
@potiuk Added fix for flag `upgrade_to_newer_dependencies` :)

**Comment by Bowrna** [comment]
> > > @potiuk i want to know how we can pass the `CI` flag in python? In bash we set it as env var. Do we need to pass that as flag when we rewrite in python?
> > > https://github.com/apache/airflow/blob/c58b5e583bcad83048fdee72bc9ddf6c3a7b4229/scripts/ci/libraries/_build_images.sh#L454-L463
> > 
> > 
> > @potiuk Please share your view on this when you get time.
> 
> Same as in Bash. Just add the `PIP_PROGRESS_BAR==off` if `CI` is set to `true`. There is no need to pass it further. But in order to do it nicely you can add `--ci` parameter with CI `envvar` via click.

@potiuk Added this support too

**Review by potiuk — APPROVED** [review]

File: dev/breeze/src/airflow_breeze/breeze.py [hunk 2]
```diff
@@ -181,6 +181,12 @@ def main():
     help='The basic apt runtime dependencies to use when building the images.',
     envvar='RUNTIME_APT_DEPS',
 )
+option_ci_flag = click.option(
+    '--ci',
+    help='Enabling this option will off the pip progress bar',
```
Comment (joppevos) [inline dev/breeze/src/airflow_breeze/breeze.py:186]: I assume it should be `"...will TURN off the pip..."`? :)

**Pull request merged**